src/id_mm.c

```diff
@@ -270,12 +270,12 @@ void MM_BombOnError(bool bomb)
 
 #ifndef ID_MM_DEBUGARENA
 
-typedef struct ID_MM_Arena
```

Comment:
Could you also remove the typedef from the debug version of the arena below?

Out of curiosity, does the repeated typedef cause problems with the compiler you're using? I know it's invalid C99, but since C++, C11, and gcc support it I'm curious what compilers are hanging on.